lib/connextclient/ethprovider.ts

```diff
@@ -0,0 +1,60 @@
+import { Observable, from, combineLatest } from 'rxjs';
+import { ethers, BigNumber } from 'ethers';
+import { mergeMap } from 'rxjs/operators';
+
+type OnChainTransaction = {
```

Comment:
Ethers has great TypeScript support in my experience. They don't expose a type for this?

Reply:
No, since it's a contract function.
```
readonly [key: string]: ContractFunction | any;
```

Reply:
What about [ContractTransaction](https://github.com/ethers-io/ethers.js/blob/master/packages/contracts/src.ts/index.ts#L90) or one of the types it extends from?

Reply:
Yeap, good idea. Changed to `ContractTransactions` for the transfer call, but for balance calls I'm sticking with `Observable<ethers.BigNumber>` type since it's more accurate.